### ISSUE TITLE
[YSQL] Secondary index does not support IN predicate

### ISSUE THREAD

**Issue by robertpang** [YugaByte/yugabyte-db#995]
Currently secondary index does not support IN predicate and will return empty results:

```
postgres=# create table t (k int primary key, v int);
CREATE TABLE
postgres=# create index on t (v);
CREATE INDEX
postgres=# insert into t values (1, 2);
INSERT 0 1
postgres=# insert into t values (3, 4);
INSERT 0 1
postgres=# select * from t where v = 2 or v = 4;
 k | v
---+---
 1 | 2
 3 | 4
(2 rows)

postgres=# select * from t where v in (2, 4);
 k | v
---+---
(0 rows)

postgres=# explain select * from t where v in (2, 4);
                             QUERY PLAN
--------------------------------------------------------------------
 Index Scan using t_v_idx on t  (cost=0.00..4.01 rows=1000 width=8)
   Index Cond: (v = ANY ('{2,4}'::integer[]))
(2 rows)
```

**Comment by robertpang** [comment]
Fix has been implemented in commit [be0908b](https://github.com/YugaByte/yugabyte-db/commit/be0908b45827c69ff010d9263ff865310ce722a1).

**Comment by ndeodhar** [comment]
Reopening this because IN clause still does not use the secondary index but does a scan on the base table instead. See https://github.com/YugaByte/yugabyte-db/issues/1554 for an example.
